#!/usr/bin/env python
PACKAGE = "reconfros_software"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add(                     "scale",    int_t, 0,                                                                  "Resize image by scale factor",   100,   1,  100)
gen.add(                       "dry",   bool_t, 0,                                   "Is the grass dried out near the trail (brown in the image)?",            True)
gen.add(             "difference_gb",    int_t, 0, "Lower bound for the difference of the green and blue channel for considering pixels as plants",   18,   0,  255)
gen.add(         "gauss_kernel_size",    int_t, 0,                                "Number of rows and cols for the gauss convolution (+1 if even)",    5,   3,    5)
gen.add(               "gray_thresh",    int_t, 0,                                                                  "Threshold for the gray value",   50,   0,  255)
gen.add(                "num_points",    int_t, 0,                                           "Block size for the average direction control method",   10,   1,   20)
gen.add(                 "use_morph",   bool_t, 0,   "Using dilation and erosion to fill holes in the trail and delete fragments from the borders",            True)
gen.add( "morph_kernel_size_closing",    int_t, 0,                             "Number of rows and cols for the closing morph kernel (+1 if even)",    7,   1,  100)
gen.add( "morph_kernel_size_opening",    int_t, 0,                             "Number of rows and cols for the opening morph kernel (+1 if even)",    19,   1,  100)

exit(gen.generate(PACKAGE, "final_pipeline", "PIPELINE"))